config/cmake/HDFMacros.cmake

```diff
@@ -401,7 +401,7 @@ macro (HDF_DIR_PATHS package_prefix)
       set (${package_prefix}_INSTALL_DATA_DIR ${CMAKE_INSTALL_DATADIR})
     endif ()
     if (NOT ${package_prefix}_INSTALL_CMAKE_DIR)
-      set (${package_prefix}_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake)
+      set (${package_prefix}_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/hdf5)
```

Comment:
Need to double check this because of past issues.

Reply:
The problem with that is, by default `cmake` does not search for config files directly in `/usr/lib(64)/cmake`; they have to be in an appropriately named sub-dir inside this directory to be seen. The cmake manual lists possible paths where it does search for these files: https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure

Reply:
Right, and back in 2022-2023 timeframe this wasdiscussed, changed,discussed and changed many times. The current settings were deemed to be correct on the variety of platforms supported. 
While you may have found a case that is incorrect, based on that past history, we can't accept that single change.
I would encourage you to bring the problem up on the hdf5 forum, and hopefully through discussion we can arrive at a solution.

Reply:
I opened a discussion: https://github.com/HDFGroup/hdf5/discussions/5364. I can send a separate PR if we agree to that RFC later if needed.

Reply:
Thanks, I just remember the multi-year discussion to get that feature correct, but there are always exceptions.

Reply:
Sure, thanks for pointing to that. I appreciate your advice and it seems anyway a good idea to decouple the two PR's (in case we agree about a new default path in the discussion) as they relate to different issues.